src/systemctl/systemctl-set-default.c

```diff
@@ -66,7 +67,8 @@ static int determine_default(char **ret_name) {
 
                 r = bus_call_method(bus, bus_systemd_mgr, "GetDefaultTarget", &error, &reply, NULL);
                 if (r < 0)
-                        return log_error_errno(r, "Failed to get default target: %s", bus_error_message(&error, r));
+                        return log_error_errno(r, "Failed to get default target: %s",
+                                               r == -ERFKILL ? "Unit is masked" : bus_error_message(&error, r));
```

Comment:
also in case you do want to match for specific errors, please do so via the dbus error name, not the error number we return. We dbus error is a lot more expressive, and we translate it to an error number client-side. Hence if you have a DBusError structure — which you do here – then always use sd_bus_error_has_name() to check for the precise error.

That said, @yuwata is right: why doesn't the english language error string the server returns suffice here? if it's ugly, then it should be fixed *server* side, not client side like you do here